packages/core/test/insured-bridge/e2e/InsuredBridge.e2e.js

```diff
@@ -0,0 +1,113 @@
+const { ZERO_ADDRESS } = require("@uma/common");
+
+const { assert } = require("chai");
+
+const { ethers } = require("hardhat");
+const { Watcher } = require("@eth-optimism/watcher");
+const { getContractInterface, predeploys } = require("@eth-optimism/contracts");
+
+const { createLocalEthersFactory, createOptimismEthersFactory, getProviders } = require("./helpers/ArtifactsHelper");
+
+const { DEFAULT_ADMIN_KEY } = require("./helpers/OptimismConstants");
+
+// Create contract factories using L1 and L2 artifacts
+const factory__L1_ERC20 = createLocalEthersFactory("ExpandedERC20");
+const factory__L2_ERC20 = createOptimismEthersFactory("L2StandardERC20", true);
+const factory__L1StandardBridge = createOptimismEthersFactory("OVM_L1StandardBridge");
+const factory__L2StandardBridge = createOptimismEthersFactory("OVM_L2StandardBridge", true);
+
+describe("L1 <> L2 Deposit and Withdrawal", () => {
+  // Set up our RPC provider connections.
+  const { l1RpcProvider, l2RpcProvider } = getProviders();
+
+  // Set up our wallets (using a default private key with 10k ETH allocated to it). Need two wallets objects, one for
+  // interacting with L1 and one for interacting with L2. Both will use the same private key.
+  const l1Wallet = new ethers.Wallet(DEFAULT_ADMIN_KEY, l1RpcProvider);
+  const l2Wallet = new ethers.Wallet(DEFAULT_ADMIN_KEY, l2RpcProvider);
+
+  let l2AddressManager, l1Messenger, L1_ERC20, L2_ERC20, L1StandardBridge, L2StandardBridge, watcher;
+
+  before("deploy tokens and set up bridge contracts", async () => {
+    l2AddressManager = new ethers.Contract(
+      predeploys.Lib_AddressManager,
+      getContractInterface("Lib_AddressManager"),
+      l2RpcProvider
+    );
+
+    l1Messenger = new ethers.Contract(
+      await l2AddressManager.getAddress("OVM_L1CrossDomainMessenger"),
+      getContractInterface("OVM_L1CrossDomainMessenger"),
+      l1RpcProvider
+    );
+
+    // Tool that helps watches and waits for messages to be relayed between L1 and L2.
+    watcher = new Watcher({
```

Comment:
Does `Watcher` only work with `ethers` contracts out of the box? Just wondering if we can use `web3` but no big deal if not.

Reply:
yup. ethers only.